hpx/util/pool_timer.hpp

```diff
@@ -0,0 +1,114 @@
+//  Copyright (c) 2016 Bibek Wagle
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_UTIL_POOL_TIMER)
+#define HPX_UTIL_POOL_TIMER
+
+#include <hpx/config.hpp>
+#include <hpx/util/function.hpp>
+#include <hpx/lcos/local/spinlock.hpp>
+
+#include <boost/thread/locks.hpp>
+#include <boost/shared_ptr.hpp>
+#include <boost/enable_shared_from_this.hpp>
+
+#include <boost/asio.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+
+namespace hpx { namespace util
+{
+    class pool_timer;
+}}
+
+namespace hpx { namespace util { namespace detail
+{
+    ///////////////////////////////////////////////////////////////////////////
+    class HPX_EXPORT pool_timer
+      : public boost::enable_shared_from_this<pool_timer>
```

Comment:
You might want to consider to move the implementation into the source file, which would allow to remove the asio include altogether.